core/src/com/unciv/ui/trade/DiplomacyScreen.kt

```diff
@@ -174,7 +174,15 @@ class DiplomacyScreen(val viewingCiv:CivilizationInfo):CameraStageBaseScreen() {
         }
         diplomacyTable.add(giveGiftButton).row()
         if (isNotPlayersTurn()) giveGiftButton.disable()
-        
+
+
+        val improveTileButton = "Gift Improvement".toTextButton()
+        improveTileButton.onClick {
+            rightSideTable.clear()
+            rightSideTable.add(ScrollPane(getImprovementGiftTable(otherCiv)))
+        }
+
+        diplomacyTable.add(improveTileButton).row()
```

Comment:
Don't forget to add `if (isNotPlayersTurn()) improveTileButton.disable()`, otherwise you could interact with city-states in this way when it is not your turn

Reply:
i did add it (line 176)

Reply:
nevermind

Reply:
added